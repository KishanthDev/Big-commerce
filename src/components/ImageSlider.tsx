"use client";
import React, { useState } from "react";
import { motion } from "framer-motion";
import Image from "next/image";

const images = [
  "/partner/algolia-logo.svg",
  "/partner/bloomreach-logo.svg",
  "/partner/fdx-logo.svg",
  "/partner/klaviyo-logo.svg",
  "/partner/makeswift-logo.svg",
  "/partner/paypal-logo.svg",
  "/partner/stripe-logo.svg",
];

const ImageSlider = () => {
  const [hover, setHover] = useState(false);

  return (
    <div
      data-testid="hover-slider"
      className="overflow-hidden w-full py-8 relative"
      onMouseEnter={() => setHover(true)}
      onMouseLeave={() => setHover(false)}
    >
      <div className="w-full flex overflow-hidden relative">
        <motion.div
          className="flex space-x-6 min-w-max"
          animate={{ x: ["0%", "-100%"] }}
          transition={{
            ease: "linear",
            duration: hover ? 30 : 50,
            repeat: Infinity,
          }}
        >
          {images.concat(images).map((src, index) => (
            <div
              key={index}
              className="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
            >
              <Image src={src} alt={src} width={50} height={50} className="object-contain" />
            </div>
          ))}
        </motion.div>
      </div>
    </div>
  );
};

export default ImageSlider;
