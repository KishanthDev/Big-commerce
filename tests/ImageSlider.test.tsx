import { render, screen, fireEvent } from '@testing-library/react';
import ImageSlider from '@/components/ImageSlider';



describe('ImageSlider Component', () => {
  test('should render images correctly', () => {
    render(<ImageSlider />);

    const imageElements = screen.getAllByRole('img');
    expect(imageElements.length).toBe(12);
  });

  test('should change animation duration on hover', () => {
    render(<ImageSlider />);

    const slider = screen.getByRole('presentation');

    expect(slider).toHaveStyle('transition-duration: 50s');

    fireEvent.mouseEnter(slider);

    expect(slider).toHaveStyle('transition-duration: 30s');

    fireEvent.mouseLeave(slider);

    expect(slider).toHaveStyle('transition-duration: 50s');
  });

  test('should match the snapshot', () => {
    const { asFragment } = render(<ImageSlider />);
    expect(asFragment()).toMatchSnapshot();
  });
});
