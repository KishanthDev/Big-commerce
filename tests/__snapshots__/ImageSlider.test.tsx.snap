// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden w-full py-8 relative"
  >
    <div
      class="w-full flex overflow-hidden relative"
    >
      <div
        class="flex space-x-6 min-w-max"
      >
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/algolia-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/algolia-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/bloomreach-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/bloomreach-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/fdx-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/fdx-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/klaviyo-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/klaviyo-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/makeswift-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/makeswift-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/paypal-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/paypal-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/stripe-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/stripe-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/algolia-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/algolia-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/bloomreach-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/bloomreach-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/fdx-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/fdx-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/klaviyo-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/klaviyo-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/makeswift-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/makeswift-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/paypal-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/paypal-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
        <div
          class="w-52 h-32 flex-shrink-0 flex items-center justify-center bg-white border border-gray-300 rounded-lg shadow-md p-3"
        >
          <img
            alt="/partner/stripe-logo.svg"
            class="object-contain"
            data-nimg="1"
            decoding="async"
            height="50"
            loading="lazy"
            src="/partner/stripe-logo.svg"
            style="color: transparent;"
            width="50"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
